frame/fast-unstake/src/types.rs

```diff
@@ -0,0 +1,119 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types used in the Fast Unstake pallet.
+
+use crate::*;
+use codec::{Decode, Encode};
+use frame_support::{
+	traits::{Currency, IsSubType},
+	RuntimeDebugNoBound,
+};
+use pallet_nomination_pools::PoolId;
+use scale_info::TypeInfo;
+use sp_runtime::transaction_validity::{InvalidTransaction, TransactionValidityError};
+use sp_staking::EraIndex;
+use sp_std::{prelude::*, vec::Vec};
+
+pub type BalanceOf<T> = <<T as pallet_staking::Config>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// An unstake request.
+#[derive(
+	Encode, Decode, Eq, PartialEq, Clone, scale_info::TypeInfo, frame_support::RuntimeDebug,
+)]
+pub struct UnstakeRequest<AccountId> {
+	/// Their stash account.
+	pub(crate) stash: AccountId,
+	/// The list of eras for which they have been checked.
+	pub(crate) checked: Vec<EraIndex>,
```

Comment:
Should we not use BoundedVec here? We could probably provide a `MaximumUnstakeRequest` and reject new requests if we have reached the limit?

Reply:
Actually, easier said than done. Technically, the `checked` will not exceed `BondingDuration`

But, assuming `BondingDuration = 28` (like Polkadot), we are almost done with checking you for entire `(0..=28)`, but era 29 starts. This means that now we care about era `(1..=29)`, but the `0` is still stored in there, so the length is more than `BondingDuration`. And while practically it is really hard, in practice this can happen multiple times.

I will think about it, but bird eye view, given that we want this pallet ASAP, I think the hassle of boudning it is not worth the complexity it brings.

Reply:
69128f951030c649833afe2ff6ac3837d5d16c5b

Reply:
69128f951030c649833afe2ff6ac3837d5d16c5b